std/math.d

```diff
@@ -4103,9 +4103,55 @@ long lrint(real x) @trusted pure nothrow @nogc
 
             return sign ? -result : result;
         }
+        else static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            long result;
+
+            // Rounding limit when casting from real(128-bit) to ulong.
+            enum real OF = 0x1P+112;
+
+            ulong* vl = cast(ulong*)(&x);
+
+            // Find the exponent and sign
+            ulong msb = vl[MANTISSA_MSB];
+            ulong lsb = vl[MANTISSA_LSB];
+            int exp = cast(int)((msb >> 48) & 0x7fff) - 0x3fff;
+            int sign = cast(int)(msb >> 63);
+            msb &= 0xffff_ffffffff;
+            msb |= 0x10000_00000000;
+
+            if (exp < 63)
+            {
+                // Adjust x and check result.
+                real j = sign ? -OF : OF;
+                x = (j + x) - j;
+                msb = vl[MANTISSA_MSB];
+                lsb = vl[MANTISSA_LSB];
+                exp = cast(int)((msb >> 48) & 0x7fff) - 0x3fff;
+                msb &= 0xffff_ffffffff;
+                msb |= 0x10000_00000000;
+
+                if (exp < 0)
+                    result = 0;
+                else if (exp < 48)
+                    result = cast(long) msb >> (48 - exp);
+                else if (exp == 48)
+                    result = cast(long) msb;
```

Comment:
Why does the `exp == 48` case need to be handled explicitly? Just check if `exp <= 48`, and let it shift by `0` in that case, as in the `ieeeExtended` block.